### ISSUE TITLE
Equivalent to JUnit's assumeThat

### ISSUE THREAD

**Issue by tsiq-karold** [twosigma/webtau#137]
There are cases where we legitimately want to skip a test.  For example, a particular bit of functionality is not yet available in certain environments.  In JUnit, one would use `assumeThat` to achieve this, would be useful to have an equivalent in webtau.

One thing to be aware of with this is that the webtau CLI return code is count failed + error + skipped therefore `assumeThat` like "failures" cannot count towards the regular skipped count.

**Comment by MykolaGolubyev** [comment]
Any ideas on a syntax?

**Comment by tsiq-karold** [comment]
A few ideas depending on how generic we want this to be:
```
scenarioIf('name', methodRefForDecidingIfToRun) { actualTest }
```
```
scenario('foo') {
    onlyForEnv('dev') {
        test some stuff
```
scenarioForEnvs('theName', 'dev', 'int') {
    test some stuff
```

Some of those would result in them getting registered as skipped.  If we don't want that, we can filp some of them like:
```
onlyForEnv('dev') {
    scenario('blah') {
        test some stuff
```

If we want to restrict how the decision is made (i.e. not generic predicate but just filtering by env for example) then I think it likely needs to be vararg and also negatable, i.e. `onlyForEnvs` and `notForEnvs`, etc.

**Comment by MykolaGolubyev** [comment]
I like the 
```
onlyForEnv('dev') {
    scenario('blah') {
        test some stuff
```

**Comment by MykolaGolubyev** [comment]
implemented with #198
